lib/langchain/llm/google_palm.rb

```diff
@@ -34,21 +34,20 @@ def initialize(api_key:, default_options: {})
     # Generate an embedding for a given text
     #
     # @param text [String] The text to generate an embedding for
-    # @return [Array] The embedding
+    # @return [Langchain::LLM::Response::GooglePalm] LLM response object with embedding as a value
     #
     def embed(text:)
-      response = client.embed(
-        text: text
-      )
-      response.dig("embedding", "value")
+      response = client.embed(text: text)
+
+      Langchain::LLM::Response::GooglePalm.new response, model: @defaults[:embeddings_model_name], type: "embedding"
```

Comment:
Why do we need to set the `type:` here? Doesn't the caller know what they should be getting back? Isn't up to them to check `.value` or `.completions` or `.embedding`?

Reply:
If caller does a single call and gets a response, they probably should. But what if you fire a bunch of requests to LLM(s) and then just push responses to the queue to deal with them later? I think that response should contain information about what type of response it is.

Reply:
I'd probably suggest that we don't solve problems we don't currently have. YAGNI.

Reply:
Well, I kinda already need to know the type within the response itself in order to properly handle data returned from the LLM.

Reply:
In [this comment](https://github.com/andreibondarev/langchainrb/pull/340#discussion_r1349752543) I was suggesting simplifying it further and getting rid of the whole `type` and `value` methods.

Reply:
If you have some specific interface for this object in mind, please share. I don't really know what you mean by "simplifying through getting rid of value method".